### PR TITLE
System.ArgumentOutOfRangeException F# snippets

### PR DESCRIPTION
**Pull request by albert-du**
## Summary
Adds F# code snippets for ArgumentOutOfRangeException

### CODE REVIEW

**Comment by albert-du** [comment]
@dsyme could you review this?

**Comment by albert-du** [comment]
@baronfel @dsyme Could you review?

**Review by dsyme — APPROVED** [review]

**Comment by albert-du** [comment]
@BillWagner ready for merge

**Review by BillWagner — APPROVED** [review]
LGTM

File: xml/System/ArgumentOutOfRangeException.xml [hunk 1]
```diff
@@ -83,41 +83,49 @@ The conditions in which an <xref:System.ArgumentOutOfRangeException> exception i
...
+      :::code language="fsharp" source="~/samples/snippets/fsharp/VS_Snippets_CLR_System/System.ArgumentOutOfRangeException/fs/NoElements2.fs" id="Snippet14":::
       :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_System/System.ArgumentOutOfRangeException/vb/NoElements2.vb" id="Snippet14":::
 
    3. You're attempting to retrieve an item whose index is negative. This usually occurs because you've searched a collection for the index of a particular element and have erroneously assumed that the search is successful. In the following example, the call to the <xref:System.Collections.Generic.List%601.FindIndex%28System.Predicate%7B%600%7D%29?displayProperty=nameWithType> method fails  to find a string equal to "Z" and so returns -1. However, this is an invalid index value.
 
       :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/System.ArgumentOutOfRangeException/cs/BadSearch.cs" interactive="try-dotnet" id="Snippet6":::
+      :::code language="fsharp" source="~/samples/snippets/fsharp/VS_Snippets_CLR_System/System.ArgumentOutOfRangeException/fs/BadSearch.fs" interactive="try-dotnet" id="Snippet6":::
```
Comment (gewarren) [inline xml/System/ArgumentOutOfRangeException.xml:110]: I'm guessing this was a copy/paste error, but F# doesn't support Try.NET. I will remove this in a separate PR.

**Pull request merged**